File: Snooper/src/main/java/com/prateekj/snooper/formatter/ResponseFormatterFactory.java
package com.prateekj.snooper.formatter;

public class ResponseFormatterFactory {

  public ResponseFormatter getFor(String data) {
    if(isXmlType(data)) {
      return new XmlFormatter();
    }
    if (isJsonType(data)) {
      return new JsonResponseFormatter();
    }
    return new PlainTextFormatter();
  }

  private boolean isXmlType(String data) {
    return data.toLowerCase().contains("xml");
  }
  private boolean isJsonType(String data) {
    return data.toLowerCase().contains("json");
  }
}

File: Snooper/src/main/java/com/prateekj/snooper/formatter/ResponseFormatterFactory.kt
package com.prateekj.snooper.formatter

class ResponseFormatterFactory {

  fun getFor(data: String): ResponseFormatter {
    return when {
      isXmlType(data) -> XmlFormatter()
      isJsonType(data) -> JsonResponseFormatter()
      else -> PlainTextFormatter()
    }
  }

  private fun isXmlType(data: String): Boolean {
    return data.toLowerCase().contains("xml")
  }

  private fun isJsonType(data: String): Boolean {
    return data.toLowerCase().contains("json")
  }
}
